core/pta/scmi.c

```diff
@@ -97,6 +99,41 @@ static TEE_Result cmd_process_smt_message(uint32_t ptypes,
 	return TEE_ERROR_NOT_SUPPORTED;
 }
 
+static TEE_Result cmd_process_msg_channel(uint32_t ptypes,
+					  TEE_Param params[TEE_NUM_PARAMS])
+{
+	TEE_Result res = TEE_ERROR_GENERIC;
+	const uint32_t exp_pt = TEE_PARAM_TYPES(TEE_PARAM_TYPE_VALUE_INPUT,
+						TEE_PARAM_TYPE_MEMREF_INPUT,
+						TEE_PARAM_TYPE_MEMREF_OUTPUT,
+						TEE_PARAM_TYPE_NONE);
+	unsigned int channel_id = params[0].value.a;
+	void *in_buf = params[1].memref.buffer;
+	size_t in_size = params[1].memref.size;
+	void *out_buf = params[2].memref.buffer;
+	size_t out_size = params[2].memref.size;
+
+	if (ptypes != exp_pt || !in_buf || !out_buf)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	if (IS_ENABLED(CFG_SCMI_MSG_SHM_MSG)) {
+		struct scmi_msg_channel *channel = NULL;
+
+		channel = plat_scmi_get_channel(channel_id);
+		if (!channel)
```

Comment:
This check is done in `scmi_entry_msg()` too.

Reply:
true, i'll drop the test done here.